examples/client/tui/commands.nim

```diff
@@ -722,6 +722,43 @@ proc command*(self: Tui, command: SendMessage) {.async, gcsafe, nimcall.} =
   else:
     asyncSpawn self.client.sendMessage(command.message)
 
+# Call ----------------------------------------------------------------
+
+proc help*(T: type CallRpc): HelpText =
+  let command = "call"
+  HelpText(command: command, aliases: aliased[command], parameters: @[
+    CommandParameter(name: "method", description: "RPC method"),
+    CommandParameter(name: "params", description: "Array of parameters")
+    ], description: "Calls an Ethereum RPC method")
+
+proc new*(T: type CallRpc, args: varargs[string]): T =
+  var rpcMethod = if args.len > 0: args[0] else: ""
+  var params = newJArray()
+  if args.len > 1:
+    try:
+      params = args[1].parseJson()
+    except:
+      params = newJNull()
+
+  T(rpcMethod: rpcMethod, params: params)
+
+proc split*(T: type CallRpc, argsRaw: string): seq[string] =
+  argsRaw.split(" ")
+
+proc command*(self: Tui, command: CallRpc) {.async, gcsafe,
+  nimcall.} =
+  try:
+    if command.rpcMethod == "":
+      self.wprintFormatError(getTime().toUnix,
+        "method cannot be blank, please provide an input.")
+    elif command.params.kind == JNull:
+      self.wprintFormatError(getTime().toUnix,
+        "params cannot be blank, please provide an input.")
```

Comment:
```suggestion
        "params cannot be blank and must be valid JSON, please provide an input.")
```

Reply:
This is because if there was a JSON parse error in `new` (ie user entered bad json), then this will be shown.